src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationHandler.cs

```diff
@@ -89,9 +89,32 @@ public abstract class RemoteAuthenticationHandler<TOptions> : AuthenticationHand
 
         protected abstract Task<AuthenticateResult> HandleRemoteAuthenticateAsync();
 
-        protected override Task<AuthenticateResult> HandleAuthenticateAsync()
+        protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
         {
-            return Task.FromResult(AuthenticateResult.Fail("Remote authentication does not support authenticate"));
+            // Most RemoteAuthenticationHandlers will have a PriorHandler, but it might not be set up during unit tests.
+            if (PriorHandler != null)
+            {
+                var authenticateContext = new AuthenticateContext(Options.SignInScheme);
+                await PriorHandler.AuthenticateAsync(authenticateContext);
+                if (authenticateContext.Accepted)
```

Comment:
Yep, here was my remark:

> IMHO, this fix seems fragile as it relies on the informational ClaimsIssuer property. If you really like this approach, I think it would be preferable to bind ClaimsIdentity.AuthenticationType with AuthenticationScheme.

That said, it's probably better than no check at all... :fearful: